File: C#/366. Find Leaves of Binary Tree.cs
/**
 * Link: https://leetcode.com/problems/find-leaves-of-binary-tree/
 * Author: Dung Nguyen Tien (Chris)
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public IList<IList<int>> FindLeaves(TreeNode root) {
        var height = GetHeight(root);
        var result = CreateResult(height);
        
        Traversal(root, result);
        
        return result;
    }
    
    public int Traversal(TreeNode node, List<IList<int>> result) {
        if (node == null) return -1;
        
        var leftLevel = Traversal(node.left, result);
        var rightLevel = Traversal(node.right, result);
        var level = Math.Max(leftLevel, rightLevel) + 1;
        
        result[level].Add(node.val);
        
        return level;
    }
    
    public List<IList<int>> CreateResult(int minHeight) {
        var res = new List<IList<int>>();
        
        for (var i = 0; i < minHeight; i++) {
            res.Add(new List<int>());
        }
        
        return res;
    }
    
    public int GetHeight(TreeNode node) {
        if (node == null) return 0;
        
        return 1 + Math.Max(GetHeight(node.left), GetHeight(node.right));
    }
}

File: Python/366. Find Leaves of Binary Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def findLeaves(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        def Traversal(node):
            if node is None:
                return 0
            height = 1 + max(Traversal(node.left), Traversal(node.right))
            
            map[height].append(node.val)
            return height
        map = defaultdict(list)
        Traversal(root)
        return map.values()
        